app/src/main/java/kth/jjve/xfran/BaseActivity.java

```diff
@@ -0,0 +1,111 @@
+package kth.jjve.xfran;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.ActionBarDrawerToggle;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.google.android.material.navigation.NavigationView;
+import com.google.firebase.auth.FirebaseAuth;
+
+public class BaseActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
+
+    DrawerLayout drawerLayout;
+    ActionBarDrawerToggle toggle;
+    NavigationView navigationView;
+    Toolbar toolbar;
+    Menu profileMenu;
+    FirebaseAuth fAuth;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.base_activity);
+
+        /*------ HOOKS ------*/
+        navigationView = findViewById(R.id.navigation_view);
+        drawerLayout = findViewById(R.id.base_drawer_layout);
+        toolbar = findViewById(R.id.toolbar_layout);
+
+        /*------ INIT ------*/
+        setSupportActionBar(toolbar);
+
+        profileMenu = navigationView.getMenu();
+        profileMenu.findItem(R.id.nav_logout).setVisible(false);
+        profileMenu.findItem(R.id.nav_profile).setVisible(false);
+
+        navigationView.bringToFront();
+        toggle = new ActionBarDrawerToggle(this, drawerLayout, toolbar,
+                R.string.nav_open, R.string.nav_close);
+        drawerLayout.addDrawerListener(toggle);
+        toggle.syncState();
+
+        navigationView.setNavigationItemSelectedListener(this);
+
+        /*----- FBASE -----*/
+        fAuth = FirebaseAuth.getInstance();
+
+        if (fAuth.getCurrentUser() != null) { // check if user is already logged in
+            profileMenu.findItem(R.id.nav_login).setVisible(false);
+            profileMenu.findItem(R.id.nav_profile).setVisible(true);
+            profileMenu.findItem(R.id.nav_logout).setVisible(true);
+        }
```

Comment:
could be made into a method to make onCreate shorter, but details

Reply:
yeah, it would indeed make the oncreate shorter. But in my eyes it would also become a bit less readable (I am not sure if it is worth it to replace only +- 4 lines)